From d57f007212116a098f246edacce07213011694d4 Mon Sep 17 00:00:00 2001
From: Jackson West <jgw@amazon.com>
Date: Mon, 24 May 2021 10:15:30 -0500
Subject: [PATCH] update coredns/corefile-migration to support 1.8.3

EKS-D 1.20-1 ships with coredns 1.8.3.  The corefile-migration package
does not currently have a release that supports 1.8.3 nor does cluster-api
or kubeadm for that matter use a version which does.  Temporarily ovewriting the version
to point to the comment upstream which added support.

Without this change, capi's webhook will deny a cluster spec with coredns 1.8.3 with
the following error:
`invalid: spec.kubeadmConfigSpec.clusterConfiguration.dns.imageTag: Forbidden: cannot migrate 
CoreDNS up to '1.8.3' from '1.8.3': start version '1.8.3' not supported`

Signed-off-by: Jackson West <jgw@amazon.com>
---
 go.mod | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/go.mod b/go.mod
index 68a2784b4..1fe203709 100644
--- a/go.mod
+++ b/go.mod
@@ -47,3 +47,5 @@ require (
 	sigs.k8s.io/kind v0.7.1-0.20200303021537-981bd80d3802
 	sigs.k8s.io/yaml v1.2.0
 )
+
+replace github.com/coredns/corefile-migration => github.com/coredns/corefile-migration v1.0.12-0.20210304171735-2bcdecd9196e
-- 
2.31.1

